src/java/fr/paris/lutece/plugins/workflow/modules/formspdf/service/HtmlToPDFGenerator.java

```diff
@@ -195,23 +283,185 @@ private void writeExportFile( Path directoryFile ) throws IOException
+                }
+                model.put( infoMarker.getMarker(), responseValueString );
+
+            }
+            if(infoMarker.getMarker().equals("url_admin_forms_response_detail"))
+            {
+                String linkMessage = I18nService.getLocalizedString( LINK_MESSAGE_BO, Locale.getDefault( ) );
+                model.put( infoMarker.getMarker(), "<a href=\""+ adminBaseUrl+"/jsp/admin/plugins/forms/ManageDirectoryFormResponseDetails.jsp?view=view_form_response_details&id_form_response=" + _formResponse.getId( ) + "\">" + linkMessage + "</a>" );
+            }
+            if(infoMarker.getMarker().equals("url_fo_forms_response_detail"))
+            {
+                String linkMessage = I18nService.getLocalizedString( LINK_MESSAGE_FO, Locale.getDefault( ) );
+                model.put( infoMarker.getMarker(), "<a href=\""+ _baseUrl+"/jsp/site/Portal.jsp?page=formsResponse&view=formResponseView&id_response=" + _formResponse.getId( ) + "\">" + linkMessage + "</a>" );
+            }
+            if(infoMarker.getMarker().equals("creation_date"))
+            {
+                String creationDate = _formResponse.getCreation( ).toLocalDateTime().toString();
+                String[] parts = creationDate.split("T");
+                String date = parts[0];
+                String time = parts[1];
+             model.put( infoMarker.getMarker(), date + " " + time );
+            }
+            if(infoMarker.getMarker().equals("update_date"))
+            {
+                String updateDate = _formResponse.getUpdate( ).toLocalDateTime().toString();
+                String[] parts = updateDate.split("T");
+                String date = parts[0];
+                String time = parts[1];
+                model.put( infoMarker.getMarker(), date + " " + time );
+            }
+            if(infoMarker.getMarker().equals("status"))
+            {
+                if(_formResponse.isPublished()) {
+                    String published = I18nService.getLocalizedString( PUBLISHED, Locale.getDefault( ) );
+                model.put( infoMarker.getMarker(), published );
+                } else {
+                    String notPublished = I18nService.getLocalizedString( NOT_PUBLISHED, Locale.getDefault( ) );
+                    model.put( infoMarker.getMarker(), notPublished );
+                }
+            }
+            if(infoMarker.getMarker().equals("update_date_status"))
+            {
+                String updateDate = _formResponse.getUpdateStatus( ).toLocalDateTime().toString();
+                String[] parts = updateDate.split("T");
+                String date = parts[0];
+                String time = parts[1];
+                model.put( infoMarker.getMarker(), date + " " + time );
+            }
+            if(infoMarker.getMarker().equals(CONSTANT_FORM_TITLE))
+            {
+                model.put( infoMarker.getMarker(), FormHome.findByPrimaryKey(_formResponse.getFormId()).getTitle());
+            }
```

Comment:
why is all this not covered in GenericFormsProvider.getProviderMarkerDescriptions(form) ?

Reply:
if(infoMarker.getMarker().equals(CONSTANT_FORM_TITLE))
            {
                System.out.println("CONSTANT_FORM_TITLE####################"+infoMarker.getValue());
                model.put( infoMarker.getMarker(), infoMarker.getValue());
            }
gave me => CONSTANT_FORM_TITLE####################null

So I included formTitle in the method instead                model.put( infoMarker.getMarker(), formTitle);